Finished Transcript 12 & 13 (#24)
status-im/pm: notes/meeting13.md

(Andy) 1st Design Review delayed by changes to navigation tab in app. Blocked until that is resolved.

### 4. Teller Network

(Iuri) First PoC done with a whole lot of UI changes to make it more usable. We are redoing a few more UI pieces so that we can get to a fully-working DApp which can support multiple users.

### 5. Network Incentivization

(Ricardo) This should be multiple things, not one.

(Oskar) It would be best to discuss further in the swarm calls the exact details of that if Andrea is not around. [link to GH or calendar invite?]

### 6. Protocol

(Oskar) Will meet at FCC and please ping me if you want to talk with specific people there.

### 7. DAO

(Ricardo) Research on topic democracy: changing from sub-topics to have sub-democracies, which are more secure. The problem is: to include new actions/topics I would have to pass a quorum. So now making it such that we can create a new democracy each time, with parent and child democracies that allow you to change ome properties without having to reach a supermajority each time.

We can create propoals without democracies right now for testing, and that's where we are.

## Better PR Process

> Document [here]() and in Discuss.

Reviewer: https://notes.status.im/C5pj8g7gQOu9Wo8PtDZsMw# and https://discuss.status.im/t/better-pull-requests-process/1044

Author: fixed

1. All PRs have tests.
2. Not promoting nightly as something for end users.
3. Reject PRs that don't have necessary tests.
4. Running automated tests when PR is marked ready.
5. Ask for 2 reviewers, and reviews done within a day.
6. Stop building the apk etc for all PRs.
7. Andrea looking into integration/headless tests.
8. Anton looking into more automated and iOS tests.

(Anton) After conversation with Jakub, we are not able to stop building the apks for every PR.

> (Pedro) Only concern is the first point. I'm not really a Clojure dev so it might make things tougher for me. Hopefully writing tests in Clojure is easy to pick it.

Reviewer 2: ```suggestion
(Pedro) Only concern is the first point. I'm not really a Clojure dev so it might make it tougher for me to contribute. Hopefully writing tests in Clojure is easy to pick up.
```

Author: fixed

[No response]

(Pedro) Just FYI, the e2e tests are already being triggered on review: that was implemented this morning.

## Nix

(Pedro) High level overview of Nix: a way of describing packages in a functional way. Read-only store located at the root of your drive, which can only be changed by admin. This gives us a lot of guarantees about that state in the machine, and allows e.g. having the nix branch running in the same machine side-by-side with develop. It works by setting up a special shell with the required environment variables that you need to build on top of the Nix packages.

I.t.o where we are right now: mobile phase done a couple of weeks ago. We were able to replace 3 different package managers with Nix. The hard part is to replace the desktop part of the dependencies, mostly because of Qt. Last week I was able to complete it for Linux, so right now we have Linux building and packaging and it's only the webview of MacOS that is giving us problems right now. Can hopefully get that unblocked fairly soon.

Once done, we can completely phase out the old way of doing things, merge the Nix branch and start using that exclusively.

> Does anyone have questions/roadblacks?

Reviewer 2: ```suggestion
Does anyone have questions/roadblocks?
```

Author: fixed

#### How did things go with the Android SDK?

(Pedro) Not installed inside of Nix atm (installed separately though), but gives us the advantage that external contributor only has to do `make setup` and we can guarantee that it will work. Android SDK not a blocker and can be fixed/included later.

#### (Jacek) Do we lock down the versions that are installed?

> Nix works by cloning a specific tag of a given repo. With brew, it's the same thing. As long as we clone specific tag or commit SHA, then you know that they are locked down. If you want a newer version, you replace the SHA. If you want a specific version of one tool and another specific version of another tool, that is hard. But we mostly want to focus on locking down the build environment here.

Reviewer 2: ```suggestion
Nix works by cloning a specific tag of a given repo. With brew, it's the same thing. As long as we clone specific tag or commit SHA, then you know that they are locked down. If you want a newer version, you replace the SHA. If you want a specific version of one tool and another specific version of another tool, that is hard. But we mostly want to focus on locking down the build environment here, not so much of handpicking tool versions.
```

Author: fixed

We could also have Status fork of the Nix repos (but not recommended). Or, in the recipe used to import Nix packages, you could possible fix that there.

**So everything uses the same version?**

Yes, basically. It must pull down exactly the same version that is committed on my machine to yours. If I want to build a commit SHA from a month ago, I am guaranteed that I will get the exact packages and tools used to build that version.

Nix is even more powerful than Conan (which I used to build Windows for desktop), so in time we should probably replace that too (though they are built on the same philosophy and way of managing packages, which is cool. Nix is just one layer up, basically.)

> (Oskar) Anything else? There is now a transcript for the Dev call (thanks to a bunty!) if there is something you are looking for.

Reviewer 2: ```suggestion
(Oskar) Anything else? There is now a transcript for the Dev call (thanks to a bounty!) if there is something you are looking for.
```

Author: fixed
